packages/integration-tests/src/process-controller/predicated_action_macros.ts

```diff
@@ -30,6 +30,12 @@ export const waitForSandboxToBecomeIdle = () =>
     'Watching for file changes...'
   );
 
+/**
+ * Reusable predicates: Wait for sandbox to indicate that it's executing hotswap deployment, i.e. "hotswapping resources:"
+ */
+export const waitForSandboxToHotswapResources = () =>
+  new PredicatedActionBuilder().waitForLineIncludes('hotswapping resources:');
```

Comment:
nit: just to be more explicit

```suggestion
export const waitForSandboxToBeginHotswappingResources = () =>
  new PredicatedActionBuilder().waitForLineIncludes('hotswapping resources:');
```